tools/autograd/gen_variable_type.py

```diff
@@ -908,12 +909,13 @@ def emit_fw_derivatives() -> List[str]:
 
             unpacked_arguments = ""
             for inp in differentiable_inputs:
+                zeros_fn = "zeros" if inplace and inp.name == "self" else "_efficientzerotensor"
```

Comment:
nit: you should not hardcode "self" here but use the mapping defined in: https://github.com/pytorch/pytorch/blob/92463573d87195f87f68f32dbb48ee497cdea182/tools/autograd/gen_inplace_or_view_type.py#L42

Reply:
Hmm, I don't completely understand this case yet. Doesn't this mapping tells us which input the output is aliased with? For inplace ops my understanding that they are always methods and the mutable input will automatically be `self` no?

Reply:
Ho true this is for views.
Not sure if inplace is always on self...

Reply:
yup @soulitzer is right, inplace ops always have `self` as the mutable input